src/objects/zcl_abapgit_rule.clas.abap

```diff
@@ -0,0 +1,122 @@
+CLASS zcl_abapgit_rule DEFINITION
+  PUBLIC
+  FINAL
+  CREATE PRIVATE .
+
+  PUBLIC SECTION.
+
+    INTERFACES zif_abapgit_rule .
+    CLASS-METHODS create
+      RETURNING
+        VALUE(ro_result) TYPE REF TO zif_abapgit_rule.
+  PROTECTED SECTION.
+  PRIVATE SECTION.
+    DATA mt_item TYPE zif_abapgit_rule~ty_items.
+    METHODS get_fields
+      IMPORTING
+        it_data          TYPE STANDARD TABLE
+      RETURNING
+        VALUE(rt_result) TYPE ddfields.
+    METHODS fill_value
+      IMPORTING
+        iv_rule  TYPE zif_abapgit_rule=>ty_fill_rule
+      CHANGING
+        cv_value TYPE any.
+ENDCLASS.
+
+
+
+CLASS zcl_abapgit_rule IMPLEMENTATION.
+
+  METHOD create.
+    CREATE OBJECT ro_result TYPE zcl_abapgit_rule.
+  ENDMETHOD.
+
+  METHOD zif_abapgit_rule~add_item.
+    INSERT is_item INTO TABLE mt_item.
+  ENDMETHOD.
+
+  METHOD zif_abapgit_rule~add_items.
+    INSERT LINES OF it_item INTO TABLE mt_item.
+  ENDMETHOD.
+
+  METHOD zif_abapgit_rule~apply_clear_logic.
+    DATA ls_item TYPE zif_abapgit_rule~ty_item.
+    DATA lt_field TYPE ddfields.
+
+    FIELD-SYMBOLS <ls_data> TYPE any.
+    FIELD-SYMBOLS <ls_field> TYPE dfies.
+    FIELD-SYMBOLS <lv_value> TYPE any.
+
+    IF mt_item IS INITIAL.
+      RETURN.
+    ENDIF.
+
+    lt_field = get_fields( ct_data ).
+    LOOP AT ct_data ASSIGNING <ls_data>.
+      LOOP AT lt_field ASSIGNING <ls_field>.
+        CLEAR ls_item.
+        READ TABLE mt_item INTO ls_item WITH KEY tabname   = iv_table
+                                                 fieldname = <ls_field>-fieldname.
+        IF ls_item-clear_field = abap_true.
+          ASSIGN COMPONENT <ls_field>-fieldname OF STRUCTURE <ls_data> TO <lv_value>.
+          CLEAR <lv_value>.
+        ENDIF.
+      ENDLOOP.
```

Comment:
instead of reading and looping over all fields, why not

```abap
loop at mt_item assigning <ls_item> where tabname = iv_table.
  assign component <ls_item>-fieldname of structure <ls_data> to <lv_value>.
  if sy-subrc = 0.
    clear <lv_value>.
  endif.
endloop.
```

Reply:
Logic changed as suggested